### PR TITLE
Set expanded/collapsed value of each legend row

### PR DESCRIPTION
**Pull request by Clebal**
Added `collapsed` property in layerConfig legend object to manage the collapsed state of the layer's legend.

```javascript
const layerConfig = {
  title: 'Store types',
  visible: true,
  showOpacityControl: true,
  opacity: 0.4,
  legend: {
    attr: 'storetype',
    collapsed: true, // false by default
    type: LEGEND_TYPES.CATEGORY,
    labels: DATA.map((data) => data.label),
    colors: DATA.map((data) => data.color),
  },
};
```

By this way, you can stay informed about what is the state of the legend and indicate the initial value.
Test added for this feature.

### CODE REVIEW

File: packages/react-redux/src/slices/cartoSlice.js [hunk 1]
```diff
@@ -78,9 +78,16 @@ export const createCartoSlice = (initialState) => {
       updateLayer: (state, action) => {
         const layer = state.layers[action.payload.id];
         if (layer)
+          // TODO: Study if we should use a deepmerge fn
           state.layers[action.payload.id] = {
             ...layer,
-            ...action.payload.layerAttributes
+            ...action.payload.layerAttributes,
+            ...(layer.legend && {
```
Comment (zbigg) [inline packages/react-redux/src/slices/cartoSlice.js:85]: i think state, we're unable to _add_ legent unless one already exists, not sure if that was intended
looks more like it should be:

```
(layer.legend || action.payload.layerAttributes.legend) &&  { legend: { ... }}
```

this version on the other hand prevents from removing legend at all

File: packages/react-widgets/src/widgets/LegendWidget.js [hunk 2]
```diff
@@ -39,12 +39,22 @@ function LegendWidget({ className, initialCollapsed }) {
...
+    );
+  };
+
   return (
     <LegendWidgetUI
       className={className}
       layers={layers}
       onChangeVisibility={handleChangeVisibility}
       onChangeOpacity={handleChangeOpacity}
+      onChangeLegendCollapsed={handleChangeLegendCollapsed}
```
Comment (zbigg) [inline packages/react-widgets/src/widgets/LegendWidget.js:57]: There is already `onCollapsedChange` passed to children, why not reuse it and call dispatch(updateLayer) only as effect response to actually local `collapsed` variable controlled by `setCollapsed` ?

File: packages/react-redux/src/slices/cartoSlice.js (hunk 1 above)
Comment (Clebal) [reply packages/react-redux/src/slices/cartoSlice.js:85]: If legend doesn't exist, it's added when `action.payload.attributes` is merged.

**Review by zbigg — APPROVED** [review]
LGTM, thanks!!!

**Review by VictorVelarde — APPROVED** [review]
👍🏻

**Pull request merged**